clarity/enhancer/multiband_compressor/crossover.py

```diff
@@ -0,0 +1,210 @@
+"""Class compute crossover filter for one crossover frequency."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy.signal import butter, lfilter, tf2zpk, zpk2tf
+
+if TYPE_CHECKING:
+    from numpy import ndarray
+
+
+class Crossover:
+    """
+    A class to compute crossover filter for two or more crossover frequencies.
+    This is based on [1]
+
+    References:
+    [1] D'Appolito, J. A. (1984, October). Active realization of multiway all-pass
+    crossover systems. In Audio Engineering Society Convention 76.
+    Audio Engineering Society.
+
+    Example:
+    >>> xover_freqs = np.array(
+    ...    [250, 500, 1000, 1500, 2000, 3000, 4000, 5000, 6000, 8000]
+    ...) * np.sqrt(2)
+    >>> xover = Crossover(xover_freqs)
+    >>> signal = np.random.randn(1000)
+    >>> filtered_signal = xover(signal)
+    >>> xover.plot_filter()
+    """
+
+    def __init__(
+        self,
+        freq_crossover: list | ndarray | int,
+        N: int = 4,
+        sample_rate: float = 44100,
+    ) -> None:
+        """Initialize the crossover filter.
+
+        Args:
+            freq_crossover (list | ndarray | int): The crossover frequency (Hz).
+            N (int): The order of the filter.
+            sample_rate (float): The sample rate of the signal (Hz).
+        """
+        if isinstance(freq_crossover, int):
+            freq_crossover = [freq_crossover]
+
+        if N != 4:
+            raise ValueError(f"The order of the filter must be 4. {N} was provided.")
+
+        self.xover_freqs = np.array(freq_crossover)
+        self.order = N
+
+        self.sample_rate = sample_rate
+
+        # Initialize the filter coefficients
+        self.bstore = np.zeros((self.order + 1, self.xover_freqs.shape[0], 2))
+        self.astore = np.zeros((self.order + 1, self.xover_freqs.shape[0], 2))
+        self.bstore_phi = np.zeros((2 * self.order + 1, self.xover_freqs.shape[0]))
+        self.astore_phi = np.zeros((2 * self.order + 1, self.xover_freqs.shape[0]))
+        self.compute_coefficients()
+
+    def compute_coefficients(self) -> None:
```

Comment:
Could be a function taking xover_freqs, order and sample rate and returning the bstore_phi and astore_phi values. This would also mean you could remove lines 62 and 63 from the constructor. i.e. their values are set directly from the return of the compute_coefficients function